### PR TITLE
Further key-value tests and refactorings

### PR DESCRIPTION
**Pull request by petebacondarwin**
See the individual commits for the details.

The refactoring of the cfetch stuff makes it possible to handle "cursors" for when lists come back in chunks.

Note that in order to use the module mocking (i.e. `mock('../cfetch/internal')`) I created an "internal" module that gets mocked. The actual API that downstream callers access is just `../cfetch` - but these then wrap the internal one with extra behaviour.

Note also that it is not allowed to have any TS type annotations in files that contain the string `ock(`!! This is because the `esbuild-jest` plugin runs a Babel transform of files that contain that string, and it cannot handle TS annotations.

### CODE REVIEW

File: packages/wrangler/package.json [hunk 1]
```diff
@@ -116,5 +120,18 @@
     "setupFilesAfterEnv": [
       "<rootDir>/src/__tests__/jest.setup.ts"
     ]
+  },
```
Comment (threepointone) [inline packages/wrangler/package.json:123]: this doesn't seem right, that's why we have esbuild-jest, to strip the types.

Comment (petebacondarwin) [reply packages/wrangler/package.json:123]: Because of this: https://github.com/aelbore/esbuild-jest/issues/57

**Comment by JacobMGEvans** [comment]
Looking good to me. Read through both commits. I also noticed some improvements to naming conventions which is a nice ergonomic change too.

**Comment by threepointone** [comment]
sorry for the delay on reviewing this, promise I'll do so tomorrow

**Review by threepointone — APPROVED** [review]
This all looks pretty reasonable. RIP cfetch :(

**Comment by petebacondarwin** [comment]
> This all looks pretty reasonable. RIP cfetch :(

Well you can rename it back to cfetch if you like 😸

**Pull request merged**